File: ui-test-fixtures/preview.ts
type PatchState = "added" | "changed" | "removed";

interface PatchRow {
  readonly id: number;
  state: PatchState;
  label: string;
}

const states: PatchState[] = ["added", "changed", "removed"];

export const rows: PatchRow[] = states.map((state, index) => ({
  id: index + 1,
  state,
  label: `${index + 1}: ${state.toUpperCase()}`,
}));

export function findRow(state: PatchState): PatchRow | undefined {
  return rows.find((row) => row.state === state);
}

File: ui-test-fixtures/preview.go
package fixture

import "strings"

type PatchState string

const (
	StateAdded   PatchState = "added"
	StateChanged PatchState = "changed"
	StateRemoved PatchState = "removed"
)

type PatchRow struct {
	Line    int
	State   PatchState
	Content string
}

func Label(row PatchRow) string {
	return strings.ToUpper(string(row.State)) + ": " + row.Content
}